sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1479,6 +1479,33 @@ class Analyzer(
       // Skip the having clause here, this will be handled in ResolveAggregateFunctions.
       case h: UnresolvedHaving => h
 
+      case agg @ (_: Aggregate | _: GroupingSets) =>
+        val resolved = agg.mapExpressions(resolveExpressionTopDown(_, agg))
+        val hasStructField = resolved.expressions.exists {
+          _.collectFirst { case gsf: GetStructField => gsf }.isDefined
+        }
+        if (hasStructField) {
+          // For struct field, it will be resolve as Alias(GetStructField, name),
+          // In Aggregate/GroupingSets this behavior will cause same struct field
+          // in aggExprs/groupExprs/selectedGroupByExprs will be resolved divided
+          // with different ExprId of Alias and replace failed when construct
+          // Aggregate in ResolveGroupingAnalytics, so we resolve duplicated struct
+          // field here with same ExprId
```

Comment:
I don't get it. `CleanupAliases` will remove aliases from the grouping expressions. Why do we hit the bug?

Reply:
> I don't get it. `CleanupAliases` will remove aliases from the grouping expressions. Why do we hit the bug?

This error happen when `ResolveGroupingAnalytics` construct Grouping Analytics Aggregation,
When expand expression, match error because of different ExprID

Reply:
ah i see, then can we remove aliases in grouping expressions here?

Reply:
> ah i see, then can we remove aliases in grouping expressions here?

Tried, No, since a lot api need param as `NamedExpression`

Reply:
But `Aggregate/GroupingSets` does not need `NamedExpression` for grouping expressions, right?

Reply:
> But `Aggregate/GroupingSets` does not need `NamedExpression` for grouping expressions, right?

But aggregate expressions need NamedExpression. 
 So you mean remove Alias and then  add Alias later when construct aggExpressions?
I am not sure if we can do like this.

Reply:
Sorry I didn't make myself clear. Can we remove aliases only in `Aggregate.groupingExpressions`?

Reply:
> Sorry I didn't make myself clear. Can we remove aliases only in `Aggregate.groupingExpressions`?

Ok， I  will try this way.

Reply:
cc @cloud-fan 
tried to remove alias of struct field in latest pr. Could you take a look.

Reply:
also ping @maropu